gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2912,6 +2912,10 @@ func predefineNow2(store Store, last BlockNode, d Decl, m map[Name]struct{}) (De
 	pkg := packageOf(last)
 	// pre-register d.GetName() to detect circular definition.
 	for _, dn := range d.GetDeclNames() {
+		if isUverseName(dn) {
```

Comment:
The comment above the `isUverseName` definition indicates this is slow. Should we cache the names in a map or something? Possibly using `go generate` to do this?

Reply:
Great catch

Reply:
I did a benchmark and it takes between 15 and 20 nanoseconds. Looking in a map takes 5 to 10 nanoseconds. I don't believe its worth adding more code for this because the cost is amortized. It is called once for every definition.